fms2_io/include/compute_global_checksum.inc

```diff
@@ -87,7 +87,8 @@ function compute_global_checksum_2d(fileobj, variable_name, variable_data, is_de
         chksum_val = mpp_chksum(buf_int64)
       endif
       deallocate(buf_int64)
-   type is (real(kind=real32))
+   type is (real(kind=real32)
+#ifdef OVERLOAD_R4)
```

Comment:
1. Why are there `#ifdef`s?
2. It looks like there is a ) that doesn't belong

Reply:
`mpp_chksum` can only handle real(kind=real32) variables if you compile with `-DOVERLOAD_R4`.

Reply:
@menzel-gfdl Then should there is a fatal called if it's not defined?

Reply:
Yes there is one:
```
  class default
    call error("unsupported type.")
end select
```

The `type is (real(kind=real32))` is inside the `#ifdef`.

Reply:
OK, it is now.  In this example, it wasn't.  Maybe there should be a comment that says that if `OVERLOAD_R4` isn't, then you will end up here.  

If you have an else, you could give an error that is more descriptive than "unsupported type".  It could say that "you are trying to use a real*4 without defining OVERLOAD_R4"

Reply:
so like?
```F90
type is (real(kind=real32))
  #ifdef OVERLOAD_R4
      ... 
  #else
   call error("you are trying to use a real*4 without defining OVERLOAD_R4")
  #endif 
type is (real(kind=real64))
```

Reply:
@uramirez8707 yes that is more descriptive than "unsupported type."  Telling the user to compile with -DOVERLOAD_R4 would be better too because it gives the user a solution to their problem.